src/helpers/buffers/ordering_sender.rs

```diff
@@ -327,6 +402,40 @@ impl OrderingSender {
     pub(crate) fn as_rc_stream(self: Arc<Self>) -> OrderedStream<Arc<Self>> {
         OrderedStream { sender: self }
     }
+
+    pub fn check_idle_and_reset(&self) -> bool {
+        #[cfg(not(debug_assertions))]
+        return false;
+        #[cfg(debug_assertions)]
+        self.state.lock().unwrap().check_idle_and_reset()
+    }
+
+    pub fn get_missing_messages(&self) -> Vec<LoggingRanges> {
```

Comment:
given that you have a wrapper now, we could move all this logging helper methods to the wrapper itself. that will provide nice encapsulation as we would not need to change send/receive buffer code at all